mlir_graphblas/src/lib/GraphBLAS/GraphBLASLowerPass.cpp

```diff
@@ -1507,10 +1509,15 @@ class LowerApplyGenericRewrite
     Value c1 = rewriter.create<arith::ConstantIndexOp>(loc, 1);
 
     // Build output with same shape as input, but possibly different output type
-    Value output = rewriter.create<graphblas::DupOp>(loc, inputTensor);
-    if (inputTensorType != outputTensorType)
-      output =
-          rewriter.create<graphblas::CastOp>(loc, outputTensorType, output);
+    Value output;
+    if (inPlace) {
+      output = inputTensor;
+    } else {
+      output = rewriter.create<graphblas::DupOp>(loc, inputTensor);
+      if (inputTensorType != outputTensorType)
```

Comment:
Should this check live outside the `inPlace` check? i.e. if someone asked for something `inplace=true`, but changed the output type, do we want to error out or do a `graphblas.cast`? The latter will create another intermediate tensor object, but it would still avoid the `graphblas.dup`.

Reply:
I like the idea of making it error out during the `verify` method. If `in_place` is true, the output and input types must be the same. What do you think of that?

Reply:
@jim22k [This commit addresses the printing issue.](https://github.com/metagraph-dev/mlir-graphblas/pull/267/commits/088e1d65285adcbfe9b88b6396547ced761f2771)

Reply:
I like your idea of using `verify` to raise an error if `in_place=True` and the output type is different. Can you add that code? I don't see any changes to GraphBLASOps.cpp here.

Reply:
https://github.com/metagraph-dev/mlir-graphblas/pull/267/commits/bb1bdd25bd54add51d84853b7eaef5650c99aceb adds that change.